src/panel.js

```diff
@@ -0,0 +1,76 @@
+import Vue from 'vue';
+
+export const PANEL_EMITTER_NAMESPACE = 'instantSearchPanelEmitter';
+export const PANEL_CHANGE_EVENT = 'PANEL_CHANGE_EVENT';
+
+export const createPanelProviderMixin = () => ({
+  props: {
+    emitter: {
+      type: Object,
+      required: false,
+      default() {
+        return new Vue({
+          name: 'PanelProvider',
+        });
+      },
+    },
+  },
+  provide() {
+    return {
+      [PANEL_EMITTER_NAMESPACE]: this.emitter,
+    };
+  },
+  data() {
+    return {
+      canRefine: true,
+    };
+  },
+  created() {
+    this.emitter.$on(PANEL_CHANGE_EVENT, value => {
+      this.updateCanRefine(value);
+    });
+  },
+  beforeDestroy() {
+    this.emitter.$destroy();
+  },
+  methods: {
+    updateCanRefine(value) {
+      this.canRefine = value;
+    },
+  },
+});
+
+export const createPanelConsumerMixin = ({ mapStateToCanRefine }) => ({
```

Comment:
maybe this can have a default like `({ mapStateToCanRefine = state => state.canRefine } = {})`?

Or is there a reason you didn't?

Reply:
The param is required so it cannot have a false positive (e.g. always return falsy value). I would prefer to keep it explicit for now until the connector always provide the `canRefine` value.